endpoints-cpo-locations/src/main/scala/com/thenewmotion/ocpi/locations/MspLocationsClient.scala

```diff
@@ -4,25 +4,37 @@ import scala.concurrent.{ExecutionContext, Future}
 import akka.http.scaladsl.HttpExt
 import akka.http.scaladsl.client.RequestBuilding._
 import akka.http.scaladsl.marshalling.ToEntityMarshaller
+import akka.http.scaladsl.unmarshalling.FromByteStringUnmarshaller
 import akka.stream.Materializer
-import com.thenewmotion.ocpi.common.{ClientObjectUri, OcpiClient}
-import com.thenewmotion.ocpi.msgs.AuthToken
+import com.thenewmotion.ocpi.common.{ClientObjectUri, ErrUnMar, OcpiClient}
+import com.thenewmotion.ocpi.msgs.{AuthToken, SuccessResp}
 import com.thenewmotion.ocpi.msgs.Ownership.Ours
 import com.thenewmotion.ocpi.msgs.v2_1.Locations._
 import cats.syntax.either._
-import com.thenewmotion.ocpi.msgs
-import spray.json.JsonFormat
 
-class MspLocationsClient(implicit http: HttpExt) extends OcpiClient {
+class MspLocationsClient(
+  implicit http: HttpExt,
```

Comment:
Jesus Christ!
Can this be put into a trait or something?

Reply:
Yeah, I don't like it either, but can't think of any other way to do this and doesn't tie it to a specific JSON lib.  Or we could just get rid of spray-json altogether ;)